File: rails/shadow_bounties/app/models/mission.rb
class Mission < ApplicationRecord
  belongs_to :hunter
  belongs_to :bounty
   
  validates :hunter_id, :bounty_id, presence: true
  validates :completed, inclusion: { in: [true, false] }

  def self.ransackable_attributes(auth_object = nil)
    ["completed", "created_at", "id", "notes", "updated_at"]
  end
end

File: next.js/first-app/src/lib/validations/mission.ts
import { z } from "zod"

export const missionSchema = z.object({
  hunterId: z.string().min(1, "Hunter is required"),

  bountyId: z.string().min(1, "Bounty is required"),

  notes: z
    .string()
    .min(5, "Notes must be at least 5 characters"),
})